### ISSUE TITLE
PUT Copy shows target key when source is not found [JIRA: RCS-215]

### ISSUE THREAD

**Issue by shino** [basho/riak_cs#1169]
Expected (result to AWS S3)

```
% s3curl.pl --id s3 -- -X PUT -H 'Content-Length: 0' \
    -H 'x-amz-copy-source: shino.shun.test1/aaa' -s \
    http://shino.shun.test1.s3.amazonaws.com/bbb X
<?xml version="1.0"?>
<Error>
  <Code>NoSuchKey</Code>
  <Message>The specified key does not exist.</Message>
  <Key>aaa</Key>
  <RequestId>74EE50EC9996EA4F</RequestId>
  <HostId>ilJOTHXjPIQRLheqg0495U+5r3nEbOy/xb80FQTAE6A7TG8FG6wbx50KjwLDZsRHLb6EKpfCeoU=</HostId>
</Error>
```

CS (at commit 5f6caea6d9ff68dec6e70407c56b463a434c7694)

```
% s3curl.pl --id cs -- -X PUT -H 'Content-Length: 0' \
     -H 'x-amz-copy-source: test/aaa' -s -x 127.0.0.1:15018 \
     http://test.s3.amazonaws.com/bbb X
<?xml version="1.0" encoding="UTF-8"?>
<Error>
  <Code>NoSuchKey</Code>
  <Message>The specified key does not exist.</Message>
  <Resource>/test/bbb</Resource>
  <RequestId/>
</Error>
```

**Comment by shino** [comment]
Other samples of AWS S3 responses.

Non existent source bucket:

```
% s3curl.pl --id shino -- -X PUT -H 'Content-Length: 0' \
    -H 'x-amz-copy-source: shino.shun.test2/aaa' -s \
    http://shino.shun.test1.s3.amazonaws.com/bbb X
<?xml version="1.0"?>
<Error>
  <Code>NoSuchBucket</Code>
  <Message>The specified bucket does not exist</Message>
  <BucketName>shino.shun.test2</BucketName>
  <RequestId>FAAB2420E07E1C49</RequestId>
  <HostId>Tq3IWK9nM7uz1+E1MbWRppg/hIKr4USPakZCd7UVLgw3mRIArzlIalJEmdTDxGuK9eHV6z6ZgYE=</HostId>
</Error>
```

Non existent target bucket:

```
% s3curl.pl --id shino -- -X PUT -H 'Content-Length: 0' \
     -H 'x-amz-copy-source: shino.shun.test1/aaa' -s \
     http://shino.shun.test-target.s3.amazonaws.com/bbb X
<?xml version="1.0"?>
<Error>
  <Code>NoSuchBucket</Code>
  <Message>The specified bucket does not exist</Message>
  <BucketName>shino.shun.test-target</BucketName>
  <RequestId>E655030FD29BD3AE</RequestId>
  <HostId>lVFOgbtXYy2K68HOkfCJiCfntQ892rj+1mFiHERaj+Jp2cS9E1+oD4QX2HghwD4J</HostId>
</Error>
```

**Comment by kuenishi** [comment]
Any bad consequent behaviour on specific client library or tools?

**Comment by shino** [comment]
AFAIK, no. It just fooled me when I was writing riak_test :sweat_smile:

**Comment by shino** [comment]
#1200 will address wrong resource value. This issue will be closed after it is merged. More compatibility with S3 XML structure will be another issue, if it is actually an issue.

**Comment by Basho-JIRA** [comment]
For release note:

Before this fix, error response of PUT Copy API call shows target resource path when source is not found or not accessible by the request user.  After this fix, it shows source path appropriately.

**_[posted via JIRA by Shunichi Shinohara]**_

**Comment by shino** [comment]
Addressed by #1200
